### PR TITLE
ProcessList native resource

### PR DESCRIPTION
**Pull request by anmenaga**
# PR Summary

Cross-platform native DSCv3 resource that returns running processes for the current user.
This resource will be used in testing `Export` functionality, so it is pretty limited at this point.

## PR Context

Example on Windows:
```
C:\DSCv3\bin\debug\process.exe list
{"pid":29016,"name":"Microsoft.SharePoint.exe","cmdline":"\"C:\\Program Files\\Microsoft OneDrive\\23.174.0820.0001\\Microsoft.SharePoint.exe\""}
{"pid":7332,"name":"smartscreen.exe","cmdline":"C:\\Windows\\System32\\smartscreen.exe -Embedding"}
{"pid":27660,"name":"Notepad.exe","cmdline":"\"C:\\Program Files\\WindowsApps\\Microsoft.WindowsNotepad_11.2306.15.0_x64__8wekyb3d8bbwe\\Notepad\\Notepad.exe\" "}
{"pid":10192,"name":"process.exe","cmdline":"\"C:\\DSCv3\\bin\\debug\\process.exe\" list"}
...
```

Example on Linux:
```
PS /home/andrew/DSC-3> /home/andrew/DSC-3/bin/debug/process list
{"pid":55140,"name":"pulseaudio","cmdline":"/usr/bin/pulseaudio --daemonize=no --log-target=journal"}
{"pid":55192,"name":"gvfsd-fuse","cmdline":"/usr/libexec/gvfsd-fuse /run/user/1000/gvfs -f -o big_writes"}
{"pid":55320,"name":"gvfsd-metadata","cmdline":"/usr/libexec/gvfsd-metadata"}
{"pid":55341,"name":"pwsh","cmdline":"./PS734/pwsh"}
{"pid":69967,"name":"process","cmdline":"/home/andrew/DSC-3/bin/debug/process list"}
...
```

### CODE REVIEW

File: process/process.dsc.resource.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Returns information about running processes.",
+    "type": "Microsoft/ProcessList",
+    "version": "0.1.0",
+    "get": {
+        "executable": "process",
+        "args": [
+          "get"
+        ]
+    },
+    "set": {
```
Comment (SteveL-MSFT) [inline process/process.dsc.resource.json:12]: Does `set` work?  If not, don't have it declared.

File: build.ps1 [hunk 2]
```diff
@@ -91,7 +91,7 @@ if (Test-Path $target) {
 New-Item -ItemType Directory $target > $null
 
 $windows_projects = @("pal", "ntreg", "ntstatuserror", "ntuserinfo", "registry")
-$projects = @("dsc_lib", "dsc", "osinfo", "test_group_resource", "y2j", "powershellgroup")
+$projects = @("dsc_lib", "dsc", "osinfo", "process", "test_group_resource", "y2j", "powershellgroup")
```
Comment (SteveL-MSFT) [inline build.ps1:94]: you should make it pedantic clean now

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
make it pedantic clean and fix manifest

File: process/process.dsc.resource.json (hunk 1 above)
Comment (anmenaga) [reply process/process.dsc.resource.json:12]: I could not come up with anything possible to set and implementation-wise simple enough that you can set on the actual running process.
But I need this `set` for testing. Currently it always returns successful status.

File: build.ps1 (hunk 2 above)
Comment (anmenaga) [reply build.ps1:94]: Fixed pedantic errors.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@anmenaga in the future, you'll need to submit PRs from a fork rather than a branch from this repo

**Pull request merged**